public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -20,23 +21,38 @@ import {
   EuiTitle,
 } from '@elastic/eui';
 import { useEffectOnce } from 'react-use';
+import { MessageActions } from '../../tabs/chat/messages/message_action';
 import { IncontextInsight as IncontextInsightInput } from '../../types';
 import { getNotifications } from '../../services';
 import { HttpSetup } from '../../../../../src/core/public';
 import { SUMMARY_ASSISTANT_API } from '../../../common/constants/llm';
 import shiny_sparkle from '../../assets/shiny_sparkle.svg';
+import { UsageCollectionSetup } from '../../../../../src/plugins/usage_collection/public';
+
 
 export const GeneratePopoverBody: React.FC<{
   incontextInsight: IncontextInsightInput;
   httpSetup?: HttpSetup;
+  usageCollection?: UsageCollectionSetup;
   closePopover: () => void;
-}> = ({ incontextInsight, httpSetup, closePopover }) => {
+}> = ({ incontextInsight, httpSetup, usageCollection, closePopover }) => {
   const [summary, setSummary] = useState('');
   const [insight, setInsight] = useState('');
   const [insightAvailable, setInsightAvailable] = useState(false);
   const [showInsight, setShowInsight] = useState(false);
+  const metricAppName = 'alertSumm';
+
   const toasts = getNotifications().toasts;
 
+  const reportGeneratedMetric = usageCollection
+    ? (metric: string) => {
+        usageCollection.reportUiStats(
+          metricAppName,
+          usageCollection.METRIC_TYPE.CLICK,
+          metric + '-' + uuidv4()
+        );
+      }
+    : () => {};
```

Comment:
this can be refactor as a util functions, i see we have same piece of code in `use_feed_back.tsx`